packages/ui-components/src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,461 @@
+	@Prop({ reflect: false }) dropdownPositionOptions?: Partial<ComputePositionConfig> = DEFAULT_TIME_RANGE_DROPDOWN_POSITION_OPTIONS;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disabled?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) showCalendar?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) selectedTimeOption?: ITimePickerTime;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) relativeTimePickerOptions?: IRelativeTimePickerOption[][] = DEFAULT_RELATIVE_TIME_OPTIONS_GROUPS;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) timezones?: string[] = getTimezonesNames();
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disableTimezoneSelection?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) displayCustomizeInterval?: boolean = true;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) displayTimezoneDropdown?: boolean = true;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) calendarInputMinDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) calendarInputMaxDate?: string;
+
+	// Defines what content is being displayed
+	@State() timePickerView: ETimePickerView = ETimePickerView.RelativeTimePicker;
+	// Current selected option
+	@State() selectedTimeState: ITimePickerTime;
+	// Dropdown open state
+	@State() dropdownOpen: boolean = false;
+	// Defines the calendar initial date if needed
+	@State() calendarInitialDate: string;
+	// Apply button tooltip text
+	@State() applyButtontooltipText: string = '';
+	// Defines if calendar is locked when the user is in full view and clicked customize interval
+	@State() calendarViewLocked: boolean = false;
+	// Defines if the timezone dropdown is visible in the input wrapper
+	@State() timezoneSelectionContentVisible = false;
+
+	/** @inheritdoc */
+	@Event() timeRangeChange: EventEmitter<ITimePickerTime>;
+	/** @inheritdoc */
+	@Event() dropdownStateChange: EventEmitter<boolean>;
+	/** @inheritdoc */
+	@Event() cancelClicked: EventEmitter<CustomEvent<MouseEvent>>;
+
+	@Watch('selectedTimeOption')
+	handleSelectTimeStateChange(timeState: ITimePickerTime) {
+		this.selectedTimeState = timeState;
+	}
+
+	componentWillLoad() {
+		if (isEmpty(this.selectedTimeOption)) {
+			this.selectedTimeState = {
+				key: DEFAULT_SELECTED_TIME_KEY,
+				range: getLast24HoursRange(),
+				timezone: this.getSelectedTimezone()
+			};
+
+			return;
+		}
+
+		if (!isEmpty(this.selectedTimeOption) && isEmpty(this.selectedTimeState)) {
+			this.selectedTimeState = this.selectedTimeOption;
+		}
+	}
+
+	private getSelectedTimezone = (): ITimePickerTimezone | undefined => {
+		if (this.selectedTimeState !== undefined) {
+			return this.selectedTimeState?.timezone;
+		}
+
+		const defaultTimezone = getDefaultTimezone();
+		if (this.timezones.includes(defaultTimezone)) {
+			return {
+				name: defaultTimezone,
+				offset: getTimezoneOffset(defaultTimezone)
+			};
+		}
+
+		return undefined;
+	};
+
+	private onDropdownChange = (event: CustomEvent<boolean>) => {
```

Comment:
Let's destruct this event and give it some proper naming.

```suggestion
	private onDropdownChange = ({ detail: isDropdownOpen }: CustomEvent<boolean>) => {
```

Reply:
and try to apply the same approach on the following functions